spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/RetryingMessageListenerAdapter.java

```diff
@@ -63,17 +66,49 @@ public RetryingMessageListenerAdapter(MessageListener<K, V> messageListener, Ret
 	}
 
 	@Override
-	public void onMessage(final ConsumerRecord<K, V> record) {
+	public void onMessage(final ConsumerRecord<K, V> record, Acknowledgment acknowledgment, Consumer<?, ?> consumer) {
 		getRetryTemplate().execute(new RetryCallback<Void, KafkaException>() {
```

Comment:
Can be replaced with Lamda:

![lambda-capable](https://cloud.githubusercontent.com/assets/1071590/24884803/2854db50-1e19-11e7-8337-5e971a641b2e.png)